gnmi/README.md

```diff
@@ -1,22 +1,37 @@
-# Fake gNMI Service (central datastore)
+# gNMI Service Reference Implementation (central datastore)
 
-`gnmit.GNMIServer` serves as a central datastore to which subscribing
-"tasks/agents" can be attached.
+`gnmi.GNMIServer` serves as the reference implementation of gNMI for lemming and the central datastore to which reconcilers can be attached.
 
-There are two ways to subscribe to the central datastore:
+## External Clients
 
-## 1 Subscribing using a `gnmit.Task` interface
+External clients use gNMI to configure and read state of the reference device.
+## Internal Reconcilers
 
-To use this method, define a function that satisfies the `gnmit.TaskRoutine`
-function signature, and register the task in [lemming.go](../lemming.go).
+Lemming is designed as an eventually consistent system where the gNMI cache is used as the central datastore. Reconcilers are attached to the gNMI cache and  are responsible reading the intended config and reconciling state.
+Reconcilers can optionally validate incoming SetRequest to prevent semantically incorrect values from being applied. For example, the interface reconciler can validate that MTU > 64 for Ethernet interfaces.
 
-See [lemming.go](../lemming.go) to see how
-[testagentlocal](testagentlocal/interface.go), an example task implementation,
-is registered.
+Reconcilers are creating by implementing the Reconciler interface, and optionally using the a reconciler Builder to simplify the creation.
 
-## 2 Subscribing using ygnmi
+### Reconciliation
 
-This method uses [`ygnmi`](https://github.com/openconfig/ygnmi) to receive
-updates from the central database.
+Reconcilers must implement the the reconciler.Reconciler interface. The reconciler interface is intentionally minimum to minimize the burden of creating reconcilers and allow maximum flexibility when creating a reconciler.
 
-See [testagent](testagent/testagent.go) for an example.
+#### Reacting to Values
+
+Reconcilers can react to values by either creating a "raw" gNMI Subscription using the client or by starting a Watch using ygnmi (optionally using WithWatch). The values are modelled in OpenConfig, using Watch is generally simpler and easier to write.
```

Comment:
Reconcilers can react to values by either calling a raw gNMI operation using the provided GNMIClient argument... (if the reconciler interface definition is copied here it would be plain to see).